modules/p2/ting_openlist/TingOpenlistRequestException.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/**
+ * @file
+ * The DingProviderConsentException class.
+ */
+
+/**
+ * Default exception.
+ */
+class TingOpenlistRequestException extends Exception {
+
+  /**
+   * Construct exception.
+   */
+  public function __construct($response) {
```

Comment:
Please add a proper DocBlock for constructor and arguments.

Also I would like to see the request as a property/argument here as well.